api/app/src/command/connected-user/get-connected-user.ts

```diff
@@ -52,3 +53,25 @@ export const getConnectedUsers = async ({
     },
   });
 };
+
+export const getConnectedUserByTokenOrFail = async (
+  provider: string,
+  str: string
+): Promise<ConnectedUser> => {
+  const connectedUser = await ConnectedUser.findOne({
+    where: {
+      providerMap: {
+        [provider]: {
+          token: {
+            [Op.like]: "%" + str + "%",
+          },
+        },
+      },
+    },
+  });
+
+  if (!connectedUser)
+    throw new NotFoundError(`Could not find connected user with str matching token`);
+
+  return connectedUser;
+};
```

Comment:
The reasoning for this is because the users Withings id is in the token & what we receive from the webhook is the user's withings id